### ISSUE TITLE
shm memory zone

### ISSUE THREAD

**Issue by bjne** [nginx/unit#843]
afaik, it does not look like any memory cache is exposed to applications through nginx-unit at the moment, tho
this could benefit a lot of applications. Is this something you are considering exposing, or are there other solutions
you would recommend?

**Comment by ac000** [comment]
memcached and redis come to mind.

**Comment by bjne** [comment]
Yes ofc, but does require and extra daemon running. unit could potentially expose this to applications directly with no protocol overhead.

**Comment by ac000** [comment]
If you're using PHP then there are things like APCu for example, other languages may have other similar modules.

Not sure we want/need to be re-inventing the wheel here...

**Comment by bjne** [comment]
Yes, just mentioning a feature that imho would be a great addition to unit, similar to uwsgi's caching framework:
https://uwsgi-docs.readthedocs.io/en/latest/Caching.html

**Comment by bes-internal** [comment]
> but does require an extra daemon running

why is this a problem at all? unitd is not a system service manager!

> uwsgi's caching framework

in this case, the purposes of using application programs overlap (both application servers), but note that unitd is not trying to become a framework for your programming language, but only to run, monitor and work on public protocols that draw a boundary between your application and the http server.

You can simply run `uwsgi --cache2 name=mycache,items=100 --socket :3031` and follow the documentation to get access both on the local machine and on the remote network.
You can also run your own wsgi application using unitd which uses internally a cache module that you write yourself or find on PyPI.

**Comment by sokil** [comment]
This idea has some reasons. Due to process isolation, Nginx Unit with PHP isolates APCu cache for all processes, thats why this cache can not be used in some scenarios, fort example for storeing metrics for prometheus
